pkgs/development/python-modules/openai/default.nix

```diff
@@ -0,0 +1,55 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pythonOlder
+
+# Python dependencies
+, openpyxl
+, pandas
+, pandas-stubs
+, requests
+, tqdm
+
+# Check dependencies
+, pytest-mock
+, pytestCheckHook
+}:
+
+buildPythonPackage rec {
+  pname = "openai";
+  version = "0.11.4";
+
+  disabled = pythonOlder "3.7.1";
+
+  # Use GitHub source since PyPi source does not include tests
+  src = fetchFromGitHub {
+    owner = "openai";
+    repo = "openai-python";
+    rev = "v${version}";
+    sha256 = "O2O4+GkyMyAxJqMNgiyPKoSXeJk0HGAst02QV6c9mJs=";
+  };
+
+  propagatedBuildInputs = [
+    openpyxl
+    pandas
+    pandas-stubs
```

Comment:
You are sure this is required at runtime?

Reply:
This is a great question. As far as I understand, mostly from this [comment](https://github.com/openai/openai-python/blob/26fbacb7ff699ed35997010edde3c0aee6599e5b/setup.py#L20) this package is only used with `mypy` to provide type hints. I confess I don’t know much about `mypy` or Python development using Nix, so I’m not sure if this should be a runtime dependency or not.

Reply:
I don't know how we generally handle this.